src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -177,18 +187,237 @@ public interface UserRepository extends JpaRepository<User, Long> {
     @Query("select distinct team.students from Team team where team.exercise.course.id = :#{#courseId} and team.shortName = :#{#teamShortName}")
     Set<User> findAllInTeam(@Param("courseId") Long courseId, @Param("teamShortName") String teamShortName);
 
+    /**
+     * Creates the specification to match the provided search term.
+     *
+     * @param searchTerm term to match
+     * @return specification used to chain database operations
+     */
+    private Specification<User> getSearchTermSpecification(String searchTerm) {
+        String extendedSearchTerm = "%" + searchTerm + "%";
+        return (root, query, criteriaBuilder) -> {
+            Predicate login = criteriaBuilder.like(root.get(User_.LOGIN), extendedSearchTerm);
+            Predicate email = criteriaBuilder.like(root.get(User_.EMAIL), extendedSearchTerm);
+            Predicate firstName = criteriaBuilder.like(root.get(User_.FIRST_NAME), extendedSearchTerm);
+            Predicate lastName = criteriaBuilder.like(root.get(User_.LAST_NAME), extendedSearchTerm);
+
+            return criteriaBuilder.or(login, email, firstName, lastName);
```

Comment:
This code does not look very nice. There is a lot of repetition. Is it not possible to write this more concise?

Reply:
I suggest to move this into a new class UserSpecs as shown in the example on https://stackoverflow.com/a/58243100

Reply:
Good idea :+1: solved in 5a1cc492bc2ccdbb0239e05f68649d73a6bb7f20 and 0b634275d7aaa550894edb28eb4ff30b7201b8e9